common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -1007,7 +1007,13 @@ def get_username(strategy, details, backend, user=None, *args, **kwargs):  # lin
         # The final_username may be empty and will skip the loop.
         # We are using our own version of user_exists to avoid possible case sensitivity issues.
         while not final_username or len(final_username) < min_length or user_exists({'username': final_username}):
-            username = short_username + uuid4().hex[:uuid_length]
+            # https://openedx.atlassian.net/browse/ENT-2824
+            suffix_block_list = ['420', '69', '666']
```

Comment:
are these all encompassing 🤔 ?  Would it be worth it to abstract this list to a constant that we could import across the project?

Reply:
+1! This list could potentially grow, and may even need to be doc'ed somewhere (or linked) at some point. This is a blacklist of suffixes I guess so yeah I can see it growing over time (unless AI lol!)

Reply:
The list is in this PR twice, so perhaps a constant someplace?  Also, other cultures have "bad luck" numbers too: https://en.wikipedia.org/wiki/List_of_bad_luck_signs

Reply:
yeah I get a feeling this is going to have to be a living list to be maintained somewhere (maybe even in a model haha). Maybe overkill, or no? Since nothing seems as good as letting the users make their own choice.

Also had another thought, does the SSO process also provide us with email? If so can we just default to emails for the first time default username, then hopefully users can change it later?

Reply:
taking a different approach - rather than adding a blocklist, lets alternate letters and numbers to avoid a meaningful substring